underfs/local/src/test/java/alluxio/underfs/local/LocalUnderFileSystemTest.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.local;
+
+import alluxio.Constants;
+import alluxio.underfs.UnderFileSystem;
+import alluxio.util.io.PathUtils;
+import alluxio.util.network.NetworkAddressUtils;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+
+/**
+ * Unit tests for the {@link LocalUnderFileSystem}.
+ */
+public class LocalUnderFileSystemTest {
+  private String mLocalUfsRoot;
+  private UnderFileSystem mLocalUfs;
+  private final File mTestDirectory = createTestingDirectory();
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  @Before
+  public void setUp() throws IOException {
+    mLocalUfsRoot = mTestDirectory.getAbsolutePath();
+    mLocalUfs = LocalUnderFileSystem.get(mLocalUfsRoot);
+  }
+
+  @After
+  public void tearDown() {
+    cleanUpOldFiles(mTestDirectory);
+  }
+
+  @Test
+  public void create() throws IOException {
+    final int size = 512;
```

Comment:
Can explain more on the scenarios that you would use final keyword for local variables?